packages/react-components/react-alert/src/components/Alert/useAlert.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import { Button } from '@fluentui/react-button';
+import {
+  CheckmarkCircle16Filled,
+  DismissCircle16Filled,
+  ErrorCircle16Filled,
+  Warning16Filled,
+} from '@fluentui/react-icons';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+
+import type { AlertProps, AlertState } from './Alert.types';
+
+/**
+ * Create the state required to render Alert.
+ *
+ * The returned state can be modified with hooks such as useAlertStyles_unstable,
+ * before being passed to renderAlert_unstable.
+ *
+ * @param props - props from this instance of Alert
+ * @param ref - reference to root HTMLElement of Alert
+ */
+export const useAlert_unstable = (props: AlertProps, ref: React.Ref<HTMLElement>): AlertState => {
+  const { intent } = props;
+
+  /** Determine the icon to render based on the intent */
+  let iconToUse;
+  switch (intent) {
+    case 'success':
+      iconToUse = <CheckmarkCircle16Filled />;
```

Comment:
```suggestion
      iconToUse = <CheckmarkCircleFilled />;
```

Can we use scalable icons? (Consumers will be able to apply styles to a slot to change their size instead of managing icons via slots)

Reply:
Updated. Are you suggesting these icons be sized via the size prop or in the styles.ts file?

Reply:
In `useAlertStyles.ts`, specifying `fontSize` should be enough. It's how it's done in `MenuItem`:

https://github.com/microsoft/fluentui/blob/07675a4abf9cf00d463e623a58846e0092910fc1/packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.ts#L62-L65